echemdb/cv/entry.py

```diff
@@ -345,6 +346,93 @@ def __repr__(self):
         """
         return f"Entry({repr(self.identifier)})"
 
+    def _normalize_field_name(self, field_name):
+        r"""
+        Return a field name when it exists in the `echemdb` resource's field names.
+
+        If 'j' is asked and does not exists in the resource's field names,
+        'I' will be checked instead.
+
+        EXAMPLES::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry._normalize_field_name('j')
+            'j'
+            >>> entry._normalize_field_name('x')
+            Traceback (most recent call last):
+            ...
+            ValueError: No axis named 'x' found.
+
+        """
+        if field_name in self.package.get_resource("echemdb").schema.field_names:
+            return field_name
+        if field_name == "j":
+            return self._normalize_field_name("I")
+        raise ValueError(f"No axis named '{field_name}' found.")
+
+    def thumbnail(
+        self, stream=False, html=False, width=2, height=1, linewidth=1, color="b"
+    ):
```

Comment:
```suggestion
    def thumbnail(
        self, stream=False, html=False, width=2, height=1, linewidth=1, color="b"
    ):
```

I believe that this interface is a bit over-engineered. (A signal here is that `html` is ignored if `stream`.)

Also, `html` is not really convenient since it does not let me add more attributes on the HTML `img` tag.

I would just always return a binary string instead.

To use this in the website, we'd just have filter `base64` (probably already exists in jinja) and then write:

```html
<img src="data:image/png;base64,{{ entry.thumbnail | base64 }}">
```

It's strange that width & height are not in pixels here.

Reply:
Agreed. As discussed if someone wants a matplotlib plot, one would anyway do that from the entries data.